### PR TITLE
fix: Windows + VMSS external svc routing broken

### PR DESCRIPTION
**Pull request by adelina-t**
The VMSSs for Linux and Windows agentpools follow a different
naming patterns. However, when populating azure.json, aks-engine
does not take that into account and we end up with the wrong name
for the primareyScaleSetName variable. Thus, the LB created for
service will have a empty backend pool, as there isn't a
VMSS with the desired name.

Properly populating the ARM variable for the primaryScaleSet will
fix this issue.

Fixes: #809

**Reason for Change**:

**Issue Fixed**:

**Requirements**:

- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Notes**:

### CODE REVIEW

**Comment by CecileRobertMichon** [comment]
@adelina-t the unit tests failed on the lint step.

**Comment by adelina-t** [comment]
Will check. Thank you!

**Comment by jackfrancis** [comment]
These test failures are real, looking into them

File: pkg/api/types.go [hunk 1]
```diff
@@ -898,12 +898,22 @@ func (p *Properties) GetNSGName() string {
 
 // GetPrimaryAvailabilitySetName returns the name of the primary availability set of the cluster
 func (p *Properties) GetPrimaryAvailabilitySetName() string {
```
Comment (jackfrancis) [inline pkg/api/types.go]: @xizhamsft this method is called to populate AKS metadata for the cluster record. It doesn't make sense that we'd have a static value for `PrimaryAvailabilitySetName` when there is no AvailabilitySet in the cluster. Does changing this method to return an empty string have any breaking effects on the AKS VMSS scenario?

File: pkg/engine/armvariables.go [hunk 2]
```diff
@@ -340,9 +335,9 @@ func getK8sMasterVars(cs *api.ContainerService) (map[string]interface{}, error)
 		} else {
 			masterVars["primaryAvailabilitySetName"] = ""
 		}
-		masterVars["primaryScaleSetName"] = ""
 		masterVars["vmType"] = "standard"
 	}
+	masterVars["primaryScaleSetName"] = cs.Properties.GetPrimaryScaleSetName()
```
Comment (jackfrancis) [inline pkg/engine/armvariables.go]: We can just set this variable using the getter now that the getter returns an empty string if the primary agent pool is not vmss.

File: pkg/engine/armvariables_test.go [hunk 3]
```diff
@@ -124,7 +124,7 @@ func TestK8sVars(t *testing.T) {
 		"nsgName":                        "[concat(variables('masterVMNamePrefix'), 'nsg')]",
 		"orchestratorNameVersionTag":     "Kubernetes:" + testK8sVersion,
 		"primaryAvailabilitySetName":     "",
-		"primaryScaleSetName":            "[concat(parameters('orchestratorName'), '-agentpool1-',parameters('nameSuffix'), '-vmss')]",
+		"primaryScaleSetName":            cs.Properties.GetPrimaryScaleSetName(),
```
Comment (jackfrancis) [inline pkg/engine/armvariables_test.go]: This is the actual fix.

Comment (jackfrancis) [reply pkg/engine/armvariables_test.go]: In other words, we're validating that the value of the ARM variable `primaryScaleSetName` derives from the canonical `primaryScaleSetName()` method, not from an ARM expression string.

File: pkg/engine/template_generator.go [hunk 4]
```diff
@@ -968,12 +968,6 @@ func (t *TemplateGenerator) getTemplateFuncMap(cs *api.ContainerService) templat
 		"GetMasterEtcdClientPort": func() int {
 			return DefaultMasterEtcdClientPort
 		},
-		"GetPrimaryAvailabilitySetName": func() string {
```
Comment (jackfrancis) [inline pkg/engine/template_generator.go]: These template layer functions are never called.

**Pull request merged**